contracts/L1ForkArbitrator.sol

```diff
@@ -0,0 +1,117 @@
+    error MissingQuestionID();
+
+    bytes32 public arbitratingQuestionId;
+    address public payer;
+
+    constructor(address _realitio, address _forkmanager, address _token) {
+        realitio = IRealityETH(_realitio);
+        forkmanager = IForkingManager(_forkmanager);
+        token = IForkonomicToken(_token);
+    }
+
+    /* solhint-disable quotes */
+    string public metadata = '{"erc20": true}';
+
+    /// @notice Return the dispute fee for the specified question. 0 indicates that we won't arbitrate it.
+    /// @dev Uses a general default, but can be over-ridden on a question-by-question basis.
+    function getDisputeFee(bytes32) public view returns (uint256) {
+        return forkmanager.arbitrationFee();
+    }
+
+    /// @notice Request arbitration, freezing the question until we send submitAnswerByArbitrator
+    /// @dev Requires payment in token(), which the UI as of early 2024 will not handle.
+    /// @param _questionId The question in question
+    /// @param _maxPrevious If specified, reverts if a bond higher than this was submitted after you sent your transaction.
+    /// NB TODO: This is payable because the interface expects native tokens. Consider changing the interface or adding an ERC20 version.
+    function requestArbitration(
+        bytes32 _questionId,
+        uint256 _maxPrevious
+    ) external payable returns (bool) {
+        if (arbitratingQuestionId != bytes32(0))
+            revert CanOnlyArbitrateOneDispute();
+        if (_questionId == bytes32(0)) revert MissingQuestionID();
+
+        uint256 fee = getDisputeFee(_questionId);
+        if (fee == 0)
+            revert TheArbitratorMustHaveSetANonZeroFeeForTheQuestion();
+
+        // First we transfer the fee to ourselves.
+        if (!token.transferFrom(msg.sender, address(this), fee))
+            revert CouldNotDeductFee();
+
+        payer = msg.sender;
+        arbitratingQuestionId = _questionId;
+
+        realitio.notifyOfArbitrationRequest(
+            _questionId,
+            msg.sender,
+            _maxPrevious
+        );
+
+        // Now approve so that the fee can be transferred right out to the ForkingManager
+        if (!token.approve(address(forkmanager), fee))
+            revert CouldNotDeductFee();
+        IForkingManager.DisputeData memory disputeData = IForkingManager
+            .DisputeData(true, address(this), _questionId);
+        IForkingManager(forkmanager).initiateFork(disputeData);
+
+        return true;
+    }
+
+    function settleChildren(
+        bytes32 lastHistoryHash,
+        bytes32 lastAnswer,
+        address lastAnswerer
+    ) public {
+        (address child1, address child2) = IForkableStructure(address(realitio))
+            .getChildren();
+        if (child1 == address(0) || child2 == address(0)) revert NotForkedYet();
+        IRealityETH(child1).assignWinnerAndSubmitAnswerByArbitrator(
+            arbitratingQuestionId,
+            bytes32(uint256(1)),
```

Comment:
I think the forks do it the other way: child1 gets the 0 assigned in the forkmanager.


see here: https://github.com/RealityETH/subjectivocracy/blob/137dc415854208bfabdd3ec2292fc6830a728cc3/contracts/L1GlobalChainInfoPublisher.sol#L74

It's not so easy to compare given that we use different notation

maybe its right as it is

Reply:
Nice catch, for now I'll align these with that so the 1st fork is no and the 2nd fork is yes. That they match the indexes. But I made an issue to make sure it's all consistent. https://github.com/RealityETH/subjectivocracy/issues/247

Reply:
Fixed in [532759b](https://github.com/RealityETH/subjectivocracy/pull/236/commits/532759beac0180315f5687431bf161e761178551)

Reply:
Scratch that, it was right the first time. Reverted in [33213bb](https://github.com/RealityETH/subjectivocracy/pull/236/commits/33213bba614cdb85115da43f9deddd5a636d636a)

First child is yes, second child is no, the ChainInfo thing also handling "neither" for the genesis chain.